### PR TITLE
[BUGFIX release] revert deletion of filter that removed deleted model…

### PR DESCRIPTION
**Pull request by runspired**
…s when flushCanonical of hasMany was called

### CODE REVIEW

File: addon/-private/system/many-array.js [hunk 1]
```diff
@@ -142,7 +142,10 @@ export default Ember.Object.extend(Ember.MutableArray, Ember.Evented, {
   },
 
   flushCanonical(isInitialized = true) {
-    let toSet = this.canonicalState;
+    // TODO make this smarter, currently its plenty stupid
+    // TODO this filtering was re-introduced as a bugfix, but seems unneeded in 2.13
+    // with the changes to internalModel cleanup in that version.
+    let toSet = this.canonicalState.filter((internalModel) => !internalModel.isDeleted());
```
Comment (runspired) [inline addon/-private/system/many-array.js]: this seems good to me

**Comment by runspired** [comment]
@bmac this is failing because we still need to bring the getOwner and _lookupFactory stuff into the release branch

**Comment by bmac** [comment]
@runspired can you rebase now. I just cherry-picked https://github.com/emberjs/data/commit/7ef8273c68d71ca20fda0a5c2c17aa1fd09a747b to the release branch.

**Comment by runspired** [comment]
done

**Pull request merged**